mm2src/coins/z_coin.rs

```diff
@@ -943,6 +965,30 @@ impl<'a> UtxoCoinBuilder for ZCoinBuilder<'a> {
     }
 }
 
+impl<'a> UtxoBlockSyncOps for ZCoinBuilder<'a> {
+    fn calculate_starting_height_from_date(&self, date: u64, current_block_height: u64) -> Result<u64, String> {
+        let buffer = self
+            .avg_blocktime()
+            .ok_or_else(|| format!("avg_blocktime not specified in {} coin config", self.ticker))?;
+        let buffer = buffer * 24;
+        let current_time_s = now_sec();
+
+        if current_time_s < date {
+            return Err(format!(
+                "{} sync_param_date must be earlier then current date",
+                self.ticker
+            ));
+        };
+
+        let secs_since_date = current_time_s - date;
+        let days_since_date = (secs_since_date / DAY_IN_SECONDS) - 1;
+        let blocks_to_sync = (days_since_date * buffer) + buffer;
+        let block_to_sync_from = current_block_height - blocks_to_sync;
```

Comment:
Make sure `current_block_height` is less than `blocks_to_sync` to prevent overflow. RPCs are external to komodefi and we shouldn't trust them blindly.

Reply:
done